### PR TITLE
Refine startup workflow of proxy

### PR DESCRIPTION
**Pull request by longjiquan**
Signed-off-by: dragondriver <jiquan.long@zilliz.com>
issue: #13065

### CODE REVIEW

File: internal/distributed/proxy/service.go [hunk 1]
```diff
@@ -90,7 +85,7 @@ func NewServer(ctx context.Context, factory msgstream.Factory) (*Server, error)
 }
 
 func (s *Server) startGrpcLoop(grpcPort int) {
-
+	s.wg.Add(1)
```
Comment (czs007) [inline internal/distributed/proxy/service.go:88]: little risk here.
If external  execute `go startGrpc()` and then immediately  call `wg.wait`

Maybe startGrpc hasn't added(1) before wg.wait succeeded

Comment (czs007) [reply internal/distributed/proxy/service.go:88]: s.wg.Add(1) should execute before startGrpcLoop

Comment (DragonDriver) [reply internal/distributed/proxy/service.go:88]: OK, got it.

**Pull request merged**